File: ConsoleApp/src/main/java/taskscheduler/TaskSchedulerConfig.java
package taskscheduler;

import org.springframework.context.annotation.AnnotationConfigApplicationContext;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;
import org.springframework.scheduling.annotation.EnableScheduling;

@Configuration
@ComponentScan
@EnableScheduling
public class TaskSchedulerConfig {

    public static void main(String[] args) {
        AnnotationConfigApplicationContext context =
                new AnnotationConfigApplicationContext(TaskSchedulerConfig.class);
    }

}

File: BasicConsoleApp/src/main/kotlin/basicconsoleapp/config/TaskSchedulerConfig.kt
package basicconsoleapp.config

import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration
import org.springframework.scheduling.annotation.EnableScheduling

/**
 * Created by ZhangJin on 2018/6/3.
 */

@Configuration
// 开启对计划任务的支持
@EnableScheduling
@ComponentScan("basicconsoleapp")
open class TaskSchedulerConfig